lib/Bridge/TolerantParser/WorseReflection/KeywordCompletor.php

```diff
@@ -0,0 +1,136 @@
+        self::CLASS_MEMBERS => [
+            // visibility
+            'public',
+            'private',
+            'protected',
+            // scope
+            'static',
+            // property
+            'var',
+            // const
+            'const',
+            // method
+            'function',
+            // trait use
+            'use',
+        ],
+    ];
+
+    /**
+     * @var array
+     */
+    private static $allKeywords = null;
+
+    public function __construct()
+    {
+        if (self::$allKeywords === null) {
+            self::$allKeywords = array_merge(
+                array_keys(TokenStringMaps::RESERVED_WORDS),
+                array_keys(TokenStringMaps::KEYWORDS)
+            );
+        }
+    }
+    /**
+    * {@inheritDoc}
+    */
+    public function complete(Node $node, TextDocument $source, ByteOffset $offset): Generator
+    {
+        $scopeName = $this->getScopeName($node, $offset);
+        
+        // PARENT TREE VISUALISATION
+        // $parents = [
+        //     get_class($node)
+        // ];
+        // $parent = $node;
+        // while (($parent = $parent->getParent()) !== null) {
+        //     $parents[] = get_class($parent);
+        // }
+        // dump(implode("\n    ->", $parents));
+
+        $keywords =
+            (isset(self::SPECIAL_SCOPES[$scopeName])) ?
+                self::SPECIAL_SCOPES[$scopeName] :
+                self::$allKeywords;
+
+        foreach ($keywords as $keyword) {
+            yield Suggestion::createWithOptions(
+                $keyword,
+                [
+                    'type' => Suggestion::TYPE_KEYWORD
+                ]
+            );
+        }
+        return true;
+    }
+
+    private function getScopeName(Node $node, ByteOffset $offset): ?string
+    {
+        $scopeName = null;
+        if (
+            ($node instanceof MemberAccessExpression || $node instanceof ScopedPropertyAccessExpression)
+            && ($node->memberName->getEndPosition() == $offset->toInt())
+        ) {
+            $scopeName = self::MEMBER_ACCESS;
+        } elseif ($node instanceof ClassMembersNode) {
+            $scopeName = self::CLASS_MEMBERS;
+        } elseif ($node instanceof StringLiteral) {
+            $scopeName = self::STRING_LITERAL;
+        } elseif (
+            $node instanceof Variable
+            && ($node->name->getEndPosition() == $offset->toInt())
+        ) {
+            $scopeName = self::VARIABLE;
+        } elseif (
+            $node instanceof CompoundStatementNode
+            && $node->getParent() instanceof AnonymousFunctionCreationExpression
+            && $node->getParent()->closeParen->getEndPosition() == ($offset->toInt() - 1)
+            && (
+                $node->getParent()->colonToken == null
+                || $node->getParent()->colonToken->getStartPosition() > $offset->toInt()
+            )
+        ) {
+            $scopeName = self::AFTER_ANONYMOUS_FUNC_PARAMS;
+        }
```

Comment:
would prefer to remove the `else`:

if ($blah) {
    return $blah;
}

if ($foo) {
   return ...;
}

return null;
```

I prefer this as it means you, as a human, can stop parsing and don't have to read the rest of the method to see if it does anything else.